src/yunohost/certificate.py

```diff
@@ -837,7 +843,10 @@ def _dns_ip_match_public_ip(public_ip, domain):
 
 def _domain_is_accessible_through_HTTP(ip, domain):
     try:
-        requests.head("http://" + ip, headers={"Host": domain})
+        requests.head("http://" + ip, headers={"Host": domain}, timeout=30)
+    except Timeout as e:
+        # XXX what should we do here? retry?
```

Comment:
If we really have a 30 sec timeout (which to me is crazy for what we are trying to do here), then we should return False.

But maybe we could have something like : 

- have a 10-15 second timeout first
- if it doesn't work, have something like "Warning : having trouble contacting domain. Retrying..."
- retry with a 30 second timeout
- return False if still timeouting

But I'm having a hard time imagining a situation where the domain was up and everything running properly, but it takes more than 30 seconds for a HEAD request to complete :/...